ui/src/dashboards/components/OverlayControls.tsx

```diff
@@ -1,11 +1,29 @@
-import React from 'react'
-import PropTypes from 'prop-types'
+import React, {SFC, MouseEvent} from 'react'
 import classnames from 'classnames'
 
-import ConfirmOrCancel from 'shared/components/ConfirmOrCancel'
+import ConfirmOrCancel from 'src/shared/components/ConfirmOrCancel'
 import SourceSelector from 'src/dashboards/components/SourceSelector'
+import {QueryConfig, Source} from 'src/types'
 
-const OverlayControls = ({
+interface SourceOption extends Source {
+  text: string
+}
+
+interface Props {
+  onCancel: () => void
+  onSave: () => void
+  isDisplayOptionsTabActive: boolean
+  onClickDisplayOptions: (
+    displayOptions: boolean
+  ) => (event: MouseEvent<HTMLLIElement>) => void
+  isSavable: boolean
+  sources: SourceOption[]
+  onSetQuerySource: (source: any) => void
```

Comment:
why is this source's type `any`? seems like type `Source` would make it more reliable?

Reply:
Thats a mistake, i meant to change that.